### ISSUE TITLE
PAW: Implement radio buttons content in questions outside of DS components, if necessary

### ISSUE THREAD

**Issue by wesrowe** [department-of-veterans-affairs/va.gov-cms#15268]
## Description
In the PAW, the <va-radio> component puts the H1 into the shadow DOM, where it cannot be read by screen readers. That's bad.

### User story
AS A Veteran
I WANT my screen reader to read me the questions in a form
SO THAT I know how to answer them.

## Details 
* DS won't have the component fixed in time to ship in PAW
* They provided an alternative, [this comment](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/2142#issuecomment-1756068371), Option F - Primitive Components:  we would need to: 
  * stop using <va-radio> 
  * Build an alternate version of the H1 / form 
    * To help with this alternative version, you may want to look at [this other ticket](https://github.com/department-of-veterans-affairs/va.gov-cms/issues/15210) that has all the custom markup we would want
  * We can still use <va-radio-option> for the radio buttons themselves

#### Previous notes
**PURPOSE:** Laura Flannery identified a gap in the current design system radio-button question component, where the question can't include bulleted lists. Let's refine when we have better visibility into timing:
- If DST has updated the component before we're ready for staging review, we'll just add the content that way
  - 10/18 a11y update: DS system will not be ready. 
- Otherwise, we'll have to implement the questions _without_ using DS components, thus incurring tech debt. If we do have to go this route, [this ticket](https://github.com/department-of-veterans-affairs/va.gov-cms/issues/15210) contains the markup that we would want to implement

### Engineering notes
The same problem exists for <va-checkbox-group> where H1 is in the shadow DOM. We will need to deal with that as well, but  will start with radio buttons.

### Quality / testing notes
A11y-wise: we need to test implementation for: 
* Screen reader
* AXE automated testing
* keyboard testing

Laura will run standard foundational tests.

## Acceptance criteria

- [x] H1 is read by screen reader for all questions, across breakpoints
- [x] Instructional text is associated to and read with the questions
- [x] Focus order for all screens is appropriate (H1 is focused first, tab to instructional text)
- [x] Verified whether checkboxes can be fixed during this work, and if not: a ticket is cut to update va-checkbox-groups.
- [x] Design review
- [x] A11y review -- noting: closing within sprint boundary may be affected, depending on when implementation is available to review

**Comment by aklausmeier** [comment]
@wesrowe @laflannery If we do go with the non-component approach, we should consider how might we contribute back to the VA Design System via this process https://design.va.gov/about/contributing-to-the-design-system/

**Comment by laflannery** [comment]
The DS will not have updated components in the timeframe that we would needs something. However, we do have a way forward here that is suggested by the DST, Option F from [this comment](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/2142#issuecomment-1756068371)

This is only suggested though and does need to be thoroughly tested by us and Collab Cycle.

**Comment by jilladams** [comment]
Note from planning: Laura flagged that there's some risk here that A11y will fail this, if any piece of a11y doesn't work after the fact. e.g if tab order, or screen readers, are confused by the exploded radio button field. 

If that happens, we'll need to abandon this approach, meaning: probably step away from the DS components and build it accessibly in straight React. If that's the case, we will regroup under a separate ticket.

**Comment by jilladams** [comment]
#15210 is a dupe, but with useful markup. Here's that ticket body for reference: 
==

## Description 
The markup/designs of the PACT Act questions was based on Income Limits (I think) but the markup for radio button groups is different. We can't do it the same because screen readers won't read out the information in the same way.

The current `<va-radio>` component has limitations that make the markup we need not possible. Therefore, we should implement the radio button questions without using  using the component - at least for now and then iterate later when the component is updated.

The basic markup should look like this:
``` 
    <h1 id="form-question">This is the question text</h1> 
    <div id="form-instructions">
       <ul>
         <li>Bahrain</li>
         <li>Iraq</li>
         <li>Kuwait</li>
         <li>Oman</li>
         <li>Qatar</li>
         <li>Saudi Arabia</li>
         <li>Somalia</li>
         <li>The United Arab Emirates (UAE)</li>
         <li>The airspace above any of these locations</li>
       </ul>
     </div>
     <fieldset aria-labelledby="form-question form-instructions">
        <span class="error" role="alert"></span>
        [<va-radio-option> components go here]
     </fieldset>
```
[Link to a Sketch file](https://www.sketch.com/s/b579a8c6-23c8-481f-bbb6-46a36839e217/a/kKLW31r) that could help (this is what we have been using to work with the Forms team and DST)

## Acceptance Criteria
- [ ] H1 heading is outside the fieldset
- [ ] H1 receives focus when navigating between screens/questions
- [ ] Both the H1 and the instructions (bullet list) is read out by screen readers as the label of the fieldset
- [ ] Error is announced when user tried to submit without selecting an answer
- [ ] User can use the arrow keys (up/down OR left/right) to navigate the radio options and they will automatically be selected (They are NOT navigated using the tab)
- [ ] User can use the space bar to select the first option
- [ ] Requires design review
- [ ] Requires accessibility review

**Comment by chriskim2311** [comment]
@laflannery I have a PR env for you to take an initial look at the updated implementation. I had to use all semantic HTML since the `va-radio-option` component needs the `va-radio` component to handle the onChange events. Let me know if you find any issues. I have some code clean up to do but the HTML should be should be good to go and the PAW is working.

http://b9576ef66da8f82de23d1a7f61abfd16.review.vetsgov-internal/

**Comment by chriskim2311** [comment]
@laflannery Thanks for the comments! I think I got all the changes in, I have a new env for you to take a look again.
http://4e4bdb80f3833f259cd813ea64c3ba42.review.vetsgov-internal/

**Comment by laflannery** [comment]
@chriskim2311 I have a couple more notes on the radio options:
- I feel like ID's can't have spaces? I just have that in the back of my head somewhere that that is not allowed. Accessibility wise it's being read fine by NVDA but I'm wondering if this might cause issues with other devices/SRs.
![image](https://github.com/department-of-veterans-affairs/va.gov-cms/assets/106678594/36cf6ce2-df26-4455-ad11-ea2fe69b0d25)
- There is a dynamic attribute being added, `aria-describedby=true`, when an option is being selected. This isn't needed, can we remove this?
  - FWIW I think the intention here might have been the `aria-checked` attribute but that isn't needed either as we are using semantic elements and the HTML `checked` worked perfectly fine here and we don't need to supplement with ARIA.

NOTE: I'm currently having doubts, after hearing this for real, about if we want that ID on the H1 I had you add (we do still need it on the instructions though for sure). I haven't made a decision yet, I need to think it over some more. How difficult would an ngrok link be to make - because I'm wondering if testing on mobile would give me more info and be able to make a decision easier.

**Comment by laflannery** [comment]
[New ticket created to update checkboxes](https://github.com/department-of-veterans-affairs/va.gov-cms/issues/15902)

**Comment by laflannery** [comment]
Noting existing issue:
- iOS Voiceover: the `<legend>` text (or any text that is acting as the legend, in this case our instructions) does not get read out automatically when the user focuses on the inputs.
  - Confirmed this is still an issue on iOS 16. Updating my phone to see if this has been fixed in iOS 17
